src/Symfony/Component/Filesystem/Filesystem.php

```diff
@@ -615,15 +615,16 @@ public function isAbsolutePath($file)
      * @param string $dir    The directory where the temporary filename will be created
      * @param string $prefix The prefix of the generated temporary filename
      *                       Note: Windows uses only the first three characters of prefix
+     * @param string $suffix The suffix of the generated temporary filename
      *
      * @return string The new temporary filename (with path), or throw an exception on failure
      */
-    public function tempnam($dir, $prefix)
+    public function tempnam($dir, $prefix, string $suffix = '')
```

Comment:
BC break ! We can't add a new argument with default value to a public method if the class or the method are not final. 
https://symfony.com/doc/current/contributing/code/bc.html#changing-classes 

See an example of how to do it correctly:
https://github.com/symfony/symfony/blob/f4f2fda81f5293d5fdef7fbc614ea527114757d5/src/Symfony/Component/Serializer/Normalizer/AbstractNormalizer.php#L319-L325